datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py

```diff
@@ -0,0 +1,304 @@
+            TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = MonitorsApi(api_client)
+        try:
+            api_instance.update_monitor(model.Id, monitor)
+        except ApiException as e:
+            LOG.error("Exception when calling MonitorsApi->update_monitor: %s\n", e)
+            return ProgressEvent.failed(HandlerErrorCode.InternalFailure, f"Exception when updating monitor {e}")
+
+    return ProgressEvent(
+        status=OperationStatus.SUCCESS,
+        resourceModel=model,
+    )
+
+
+@resource.handler(Action.DELETE)
+def delete_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = MonitorsApi(api_client)
+        try:
+            api_instance.delete_monitor(model.Id)
+        except ApiException as e:
+            LOG.error("Exception when calling MonitorsApi->delete_monitor: %s\n", e)
+            return ProgressEvent.failed(HandlerErrorCode.InternalFailure, f"Exception when deleting monitor {e}")
+
+    return ProgressEvent(
+        status=OperationStatus.SUCCESS,
+        resourceModel=model,
+    )
+
+
+@resource.handler(Action.CREATE)
+def create_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+
+    monitor = ApiMonitor()
+    if model.Message is not None:
+        monitor.message = model.Message
+    if model.Name is not None:
+        monitor.name = model.Name
+    if model.Tags is not None:
+        monitor.tags = model.Tags
+    options = build_monitor_options_from_model(model)
+    if options:
+        monitor.options = options
+    monitor.query = model.Query
+    monitor.type = ApiMonitorType(model.Type)
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = MonitorsApi(api_client)
+        try:
+            api_instance.create_monitor(monitor)
+        except ApiException as e:
+            LOG.error("Exception when calling MonitorsApi->create_monitor: %s\n", e)
+            return ProgressEvent.failed(HandlerErrorCode.InternalFailure, f"Exception when creating monitor {e}")
+
+    return ProgressEvent(
+        status=OperationStatus.SUCCESS,
+        resourceModel=model,
+    )
```

Comment:
I believe we still have to call the read handler here, otherwise you get back a model that doesn't include the read only attributes. (Same for the update_handler)

```suggestion
    return read_handler(session, request, callback_context)
```

Reply:
true, wanted to do it, forgot 😅